x-pack/packages/kbn-elastic-assistant-common/README.md

```diff
@@ -0,0 +1,20 @@
+# @kbn/elastic-assistant-common
+
+This package provides common code consumed in both the browser, i.e. the
+`packages/kbn-elastic-assistant` package, and on the server, i.e. the
+`plugins/elastic_assistant` plugin.
```

Comment:
Should we instead consider moving the package code into `plugins/elastic_assistant/public` and this directory would be `plugins/elastic_assistant/common`? Are there arguments to be made still that it should be a package?

Reply:
> Should we instead consider moving the package code into `plugins/elastic_assistant/public` and this directory would be `plugins/elastic_assistant/common`? Are there arguments to be made still that it should be a package?

It feels like moving the common code into `plugins/elastic_assistant/public` would create a circular dependency, because the `elastic_assistant`'s `x-pack/plugins/elastic_assistant/tsconfig.json` already has a reference to the package `@kbn/elastic-assistant`:

```
  "kbn_references": [
     // ...
    "@kbn/elastic-assistant", // <--existing reference
    "@kbn/elastic-assistant-common", // <-- new reference
     // ...
  ],
```

The anonymization functions moving to the new `elastic-assistant-common` are pure functions that are truly common / usable on both the client and server. (They are used on both in this PR.) Some of the helper functions may also be applicable in other contexts where anonymization is a useful feature.

It therefore feels like moving these anonymization functions to the new `elastic-assistant-common` is package prudent, because:
- It does not create a circular dependency
- The code is truly shared (by client and server) common functions, as described by the `@kbn/elastic-assistant-common`  package's  `x-pack/packages/kbn-elastic-assistant-common/kibana.jsonc`:

```json
{
  "id": "@kbn/elastic-assistant-common",
  "owner": "@elastic/security-solution",
  "type": "shared-common" // <-- truly shared by client and server
}
```
- The common functions may be applicable in other contexts that require anonymization. Given that Kibana Packages are a lighter weight abstraction for sharing code compared to Plugins (because Plugins have, for example, lifecycle), it feels prudent to make these potentially reusable functions available via the lighter weight (Package) abstraction, given they are pure functions that don't require the capabilities of a Plugin.

Most of the types imported today from `@kbn/elastic-assistant` by the `elastic_assistant` plugin may be moved to the new `@kbn/elastic-assistant-common` package (in a future PR). If that happened, it should be possible to remove the `@kbn/elastic-assistant` dependency from `x-pack/plugins/elastic_assistant/tsconfig.json`.